tests/ui.test.ts

```diff
@@ -66,4 +79,51 @@ test.describe('FieldSelect', () => {
 		await expect(page.getByRole('option')).toHaveCount(0);
 		await expect(page.getByTitle('Clear')).not.toBeVisible();
 	});
+
+	test('models are correctly grouped and sorted', async ({ page }) => {
+		const newSessionButton = page.getByTestId('new-session');
+		await page.goto('/settings');
+		await mockTagsResponse(page);
+
+		await page.getByText('Sessions', { exact: true }).click();
+		await newSessionButton.click();
+
+		// Select a model
+		await chooseFromCombobox(page, 'Available models', MOCK_API_TAGS_RESPONSE.models[1].name);
+
+		// Mock a session with a specific model
+		await mockCompletionResponse(page, MOCK_SESSION_1_RESPONSE_1);
+		await page
+			.locator('.prompt-editor__textarea')
+			.fill('Who would win in a fight between Emma Watson and Jessica Alba?');
+		await page.getByText('Run').click();
+
+		// Create another session with a different model
+		await newSessionButton.click();
+		await chooseFromCombobox(page, 'Available models', MOCK_API_TAGS_RESPONSE.models[0].name);
+		await mockCompletionResponse(page, MOCK_SESSION_1_RESPONSE_1);
+		await page
+			.locator('.prompt-editor__textarea')
+			.fill('Who would win in a fight between Emma Watson and Jessica Alba?');
+		await page.getByText('Run').click();
+
+		await newSessionButton.click();
+		const modelCombobox = page.getByLabel('Available models');
+		await modelCombobox.click();
+
+		// Check for "Recently used models" group
+		const recentlyUsedModelsGroup = page.locator('.field-combobox-group', {
+			hasText: 'Recently used models'
+		});
+		const recentlyUsedModels = recentlyUsedModelsGroup.locator('.field-combobox-item-label-option');
+		await expect(recentlyUsedModels).toHaveCount(2);
+		await expect(recentlyUsedModels.nth(0)).toHaveText(MOCK_API_TAGS_RESPONSE.models[0].name);
+		await expect(recentlyUsedModels.nth(1)).toHaveText(MOCK_API_TAGS_RESPONSE.models[1].name);
+
+		// Check for "Other models" group
+		const otherModelsGroup = page.locator('.field-combobox-group', {
+			hasText: 'Other models'
+		});
+		await expect(otherModelsGroup).toContainText(MOCK_API_TAGS_RESPONSE.models[2].name);
```

Comment:
Let's add one minor assertion here to make sure the models are not duplicated across both groups.

```suggestion
		await expect(otherModelsGroup).toHaveCount(1);
		await expect(otherModelsGroup).toContainText(MOCK_API_TAGS_RESPONSE.models[2].name);
```